cmd/operator/deploy/crds/monitoring.googleapis.com_operatorconfigs.yaml

```diff
@@ -50,6 +50,12 @@ spec:
                 enum:
                 - none
                 - gzip
+              configCompression:
```

Comment:
IMO this makes more sense as a `.features` rather than in `.collection`.

The reason being that `.collection` fields relate more to ingress/egress of metric data, not the config itself.

The fact that the config is compressed isn't really a concern of the collectors at all actually (at least not as of today).

Reply:
Another thought I had.

If we want to make this more generalizable and future-proof, where we could provide sharding down the line. Maybe something like:
```yaml
apiVersion: monitoring.googleapis.com/v1
kind: OperatorConfig
metadata:
  namespace: gmp-public
  name: config
features:
  config:
    # Eventually deprecate and have a defaulting webhook omit it.
    compression: gzip
    # Add later when/if config-reloader is obsolete.
    shardCount: 5
```

Reply:
Un-resolving for thoughts on new API structure.

Though I am happy to bow out here and let @bwplotka take over 😄

Reply:
Nice. Do we really need option at all? Always gzip is not a bad idea.

Another point is that we will have a harder time to implement https://github.com/prometheus/prometheus/issues/9783#issuecomment-1669341273 if we depend on gzip. But so for any other sharding solution.

Reply:
Thanks for the answer on https://github.com/GoogleCloudPlatform/prometheus-engine/pull/571#issuecomment-1725867848

Reply:
We could omit it and have it transparently always-on as well.

Per rollout policy, this could be considered a blanket "performance improvement" as it allows managed collection to scale better.

However, presumably CPU usage of the config-reloaders would increase. Maybe it's worth doing a quick benchmark to see what the cost is for a relatively large configmap?